sdkms/common.go

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package sdkms
+
+import "fmt"
+
+type HyperHttpMthod string
+
+// Common HTTP methods.
+//
+// Unless otherwise noted, these are defined in RFC 7231 section 4.3.
+const (
+	MethodGet     HyperHttpMthod = "GET"
+	MethodHead    HyperHttpMthod = "HEAD"
+	MethodPost    HyperHttpMthod = "POST"
+	MethodPut     HyperHttpMthod = "PUT"
+	MethodPatch   HyperHttpMthod = "PATCH" // RFC 5789
+	MethodDelete  HyperHttpMthod = "DELETE"
+	MethodConnect HyperHttpMthod = "CONNECT"
+	MethodOptions HyperHttpMthod = "OPTIONS"
+	MethodTrace   HyperHttpMthod = "TRACE"
+)
+
+type RemovableInterface interface {
+}
+
+func ParseCustomMetadata(v map[string][]string, cus map[string]string) {
+	for k, val := range cus {
+		v[fmt.Sprintf("custom_metadata.%s", k)] = []string{val}
+	}
+}
+
+type Base64UrlSafe string
```

Comment:
this should be `[]byte` instead, and it should make sure to serialize/deserialize using base64 URL-safe character set.